domain-management/src/main/java/org/jboss/as/domain/management/security/UserDomainCallbackHandler.java

```diff
@@ -196,11 +210,35 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
                     SECURITY_LOGGER.tracef("User '%s' not found.", userName);
                     throw new UserNotFoundException(userName);
                 }
-                String password = user.require(PASSWORD).asString();
-                ((PasswordCallback) current).setPassword(password.toCharArray());
+                char[] password = new char[0];
+                if (user.hasDefined(PASSWORD)) {
+                    password = user.require(PASSWORD).asString().toCharArray();
+                }
+                SECURITY_LOGGER.tracef("User '%s' has the password '%'.",user, resolvePassword(userCredentialSourceSupplier, password));
```

Comment:
Bad for two reasons.
1 - resolvePassword will be called regardless of if trace logging is enabled.
2 - It will lead to a bug report when clear passwords are found in logs ;-)